shell/platform/darwin/ios/framework/Source/FlutterViewController.mm

```diff
@@ -164,7 +165,6 @@ - (void)performCommonViewControllerInitialization {
 
   [self setupNotificationCenterObservers];
 
-  [self connectToEngineAndLoad];
```

Comment:
I was considering that too, and as far as I can see it doesn't. The point of moving the call is to be able to actually set the initial route. If the call stays here, it is part of the FlutterViewController initialization and then it's too late to set the initial route afterwards.